cmd/launcher/main.go

```diff
@@ -62,22 +69,31 @@ func main() {
 	// fork-bombing itself. This is an ENV, because there's no
 	// good way to pass it through the flags.
 	if !env.Bool("LAUNCHER_SKIP_UPDATES", false) && !inBuildDir {
-		runNewerLauncherIfAvailable(ctx, systemSlogger.Logger)
+		if err := runNewerLauncherIfAvailable(ctx, systemSlogger.Logger); err != nil {
+			return fmt.Errorf("running newer version of launcher: %w", err)
```

Comment:
Is there a place we can log this?

Reply:
Almost all the errors in `runNewerLauncherIfAvailable` are already logged, but I see one that's not -- I'll get that updated 🙂

Reply:
Mostly it feels weird to return errors if we're just going to throw them away. So I think something is slightlt weird in the runLauncher idiom here

Reply:
I thought about returning an exit code int instead -- does that feel like a cleaner return?

Reply:
(I'd decided against it originally because returning an error felt more idiomatically Go, but 🤷‍♀️ )

Reply:
Might be cleaner to return an exit code -- then it's more clear that the usual go pattern of returning an error won't work

Reply:
That makes sense, I'll update and see how it looks!

Reply:
Yup, it's way better, thank you for the suggestion!